gtk/src/file_chooser_dialog.rs

```diff
@@ -94,3 +102,727 @@ impl FileChooserDialog {
+        self
+    }
+
+    pub fn receives_default(mut self, receives_default: bool) -> Self {
+        self.receives_default = Some(receives_default);
+        self
+    }
+
+    pub fn sensitive(mut self, sensitive: bool) -> Self {
+        self.sensitive = Some(sensitive);
+        self
+    }
+
+    pub fn tooltip_markup(mut self, tooltip_markup: &str) -> Self {
+        self.tooltip_markup = Some(tooltip_markup.to_string());
+        self
+    }
+
+    pub fn tooltip_text(mut self, tooltip_text: &str) -> Self {
+        self.tooltip_text = Some(tooltip_text.to_string());
+        self
+    }
+
+    pub fn valign(mut self, valign: Align) -> Self {
+        self.valign = Some(valign);
+        self
+    }
+
+    pub fn vexpand(mut self, vexpand: bool) -> Self {
+        self.vexpand = Some(vexpand);
+        self
+    }
+
+    pub fn vexpand_set(mut self, vexpand_set: bool) -> Self {
+        self.vexpand_set = Some(vexpand_set);
+        self
+    }
+
+    pub fn visible(mut self, visible: bool) -> Self {
+        self.visible = Some(visible);
+        self
+    }
+
+    pub fn width_request(mut self, width_request: i32) -> Self {
+        self.width_request = Some(width_request);
+        self
+    }
+
+    pub fn action(mut self, action: FileChooserAction) -> Self {
+        self.action = Some(action);
+        self
+    }
+
+    pub fn create_folders(mut self, create_folders: bool) -> Self {
+        self.create_folders = Some(create_folders);
+        self
+    }
+
+    pub fn do_overwrite_confirmation(mut self, do_overwrite_confirmation: bool) -> Self {
+        self.do_overwrite_confirmation = Some(do_overwrite_confirmation);
+        self
+    }
+
+    pub fn extra_widget<P: IsA<Widget>>(mut self, extra_widget: &P) -> Self {
+        self.extra_widget = Some(extra_widget.clone().upcast());
+        self
+    }
+
+    pub fn filter(mut self, filter: &FileFilter) -> Self {
+        self.filter = Some(filter.clone());
+        self
+    }
+
+    pub fn local_only(mut self, local_only: bool) -> Self {
+        self.local_only = Some(local_only);
+        self
+    }
+
+    pub fn preview_widget<P: IsA<Widget>>(mut self, preview_widget: &P) -> Self {
+        self.preview_widget = Some(preview_widget.clone().upcast());
+        self
+    }
+
+    pub fn preview_widget_active(mut self, preview_widget_active: bool) -> Self {
+        self.preview_widget_active = Some(preview_widget_active);
+        self
+    }
+
+    pub fn select_multiple(mut self, select_multiple: bool) -> Self {
+        self.select_multiple = Some(select_multiple);
+        self
+    }
+
+    pub fn show_hidden(mut self, show_hidden: bool) -> Self {
+        self.show_hidden = Some(show_hidden);
+        self
+    }
+
+    pub fn use_preview_label(mut self, use_preview_label: bool) -> Self {
+        self.use_preview_label = Some(use_preview_label);
+        self
+    }
+
+    pub fn add_button(mut self, title: &str, response_type: ResponseType) -> Self {
```

Comment:
```suggestion
    pub fn button(mut self, title: &str, response_type: ResponseType) -> Self {
```

Maybe? But if we do it for this dialog, shouldn't we do it for all the others too for consistency?

Reply:
Does it make sense to have this as part of the Builder pattern actually ?

Reply:
> Maybe? But if we do it for this dialog, shouldn't we do it for all the others too for consistency?

But then it feels like we're simply replacing the current one... We could replace with `buttons` and set them all at once though...

> Does it make sense to have this as part of the Builder pattern actually ?

You can't add buttons otherwise, so not sure how to do it if not by adding a manual thing...

Reply:
Well, you can't set a filter neither for a FileChooser. The builders set properties not wrap function calls.

You can build the object from the builder and then call the methods to add/set the buttons on it I guess ?

Reply:
There is no such method, that's the whole issue. You can only set them when creating the dialog apparently.

Reply:
What about https://gtk-rs.org/gtk3-rs/stable/0.14/docs/gtk/prelude/trait.DialogExtManual.html#tymethod.add_buttons and https://gtk-rs.org/gtk3-rs/stable/0.14/docs/gtk/prelude/trait.DialogExt.html#tymethod.add_button ?

Reply:
Ah indeed! Then there is no point for this PR. :)